google/cloud/pubsublite/internal/wire/single_partition_publisher.py

```diff
@@ -120,11 +135,25 @@ def _handle_response(self, response: PublishResponse):
                     "Received an publish response on the stream with no outstanding publishes."
                 )
             )
-        next_offset: Cursor = response.message_response.start_cursor.offset
-        batch: List[WorkItem[PubSubMessage]] = self._outstanding_writes.pop(0)
-        for item in batch:
-            item.response_future.set_result(Cursor(offset=next_offset))
-            next_offset += 1
+        ranges = response.message_response.cursor_ranges
+        ranges.sort(key=_get_start_index)
```

Comment:
since proto3 has default 0s, I think this can just be `lambda r: r.start_index`

Reply:
Yes, this works. I was getting some error before and thought this was the issue, but probably not.